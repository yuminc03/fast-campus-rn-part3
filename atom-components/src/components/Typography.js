import React from 'react';
import { Text as RNText } from 'react-native';

export class Typography extends React.Component {
  render() {
    return (
      <RNText
        style={{
          color: this.props.color,
          fontSize: this.props.fontSize,
        }}
      >
        {this.props.children}
      </RNText>
    )
  }
}

Typography.propTypes = {
  color: PropTypes.string,
  fontSize: PropTypes.number.isRequired,
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired
}