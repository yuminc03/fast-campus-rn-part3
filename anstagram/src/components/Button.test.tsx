import React from 'react';
import { render, screen } from '@testing-library/react-native';
import { Button } from "./Button";
import { Typography } from "./Typography";

test('Button Test', async () => {
  const onPressed = () => {};
  const expectedButtonName = 'TEST_BUTTON_2';
  
  render(
    <Button onPress={onPressed}>
      <Typography>{expectedButtonName}</Typography>
    </Button>
  );

  expect(screen.toJSON()).toMatchSnapshot();
});