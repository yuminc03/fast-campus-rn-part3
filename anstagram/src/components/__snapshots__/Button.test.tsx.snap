// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Test 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": undefined,
      "paddingHorizontal": undefined,
      "paddingVertical": undefined,
    }
  }
>
  <Text
    style={
      {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    TEST_BUTTON_2
  </Text>
</View>
`;
